### PR TITLE
MGMT-18981: delete ici in ibio e2e test

### PR DESCRIPTION
**Pull request by giladravid16**
Delete the ImageClusterInstall at the end of IBIO's e2e test, in order to check that IBIO handles it correctly.

closes [MGMT-18981](https://issues.redhat.com//browse/MGMT-18981)

### CODE REVIEW

**Comment by giladravid16** [comment]
/pj-rehearse pull-ci-openshift-image-based-install-operator-main-e2e-ibio

File: ci-operator/step-registry/image-based-install-operator/post/gather/image-based-install-operator-post-gather-commands.sh [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+set -xeo pipefail
+
+cd /root/dev-scripts
+source common.sh
+source utils.sh
+source network.sh
+
+export ARTIFACT_DIR=/ibio-post-gather
+mkdir -p ${ARTIFACT_DIR}
+oc get baremetalhost ostest-extraworker-0 -n openshift-machine-api -o yaml > ${ARTIFACT_DIR}/baremetalhost.yaml
```
Comment (eranco74) [inline ci-operator/step-registry/image-based-install-operator/post/gather/image-based-install-operator-post-gather-commands.sh:25]: Try to get the dataimage as well (if it fails just ignore it), if it wasn't deleted we would like to see the CR

File: ci-operator/step-registry/image-based-install-operator/post/delete-ici/image-based-install-operator-post-delete-ici-commands.sh [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+cd /root/dev-scripts
+source common.sh
+source utils.sh
+source network.sh
+
+echo "### Delete the ImageClusterInstall"
+oc delete imageclusterinstall ibi-cluster -n ibi-cluster
+oc wait -n ibi-cluster --for=delete imageclusterinstalls.extensions.hive.openshift.io/ibi-cluster --timeout=10m
+oc wait -n openshift-machine-api --for=delete dataimages.metal3.io/ostest-extraworker-0 --timeout=60m
```
Comment (eranco74) [inline ci-operator/step-registry/image-based-install-operator/post/delete-ici/image-based-install-operator-post-delete-ici-commands.sh:28]: 60 minutes seems too long, if it doesn't get deleted after 10 it won't get deleted after 60.

File: ci-operator/step-registry/image-based-install-operator/post/delete-ici/image-based-install-operator-post-delete-ici-commands.sh [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+echo "### Removing the config iso includes rebooting ibi-host, waiting until ibi-cluster to come back up"
+until oc --kubeconfig /root/ibi-cluster/kubeconfig get clusterversion,clusteroperators; do sleep 5; done
```
Comment (eranco74) [inline ci-operator/step-registry/image-based-install-operator/post/delete-ici/image-based-install-operator-post-delete-ici-commands.sh:31]: Nice validation.
Add some timeout for this as well, e.g.
```
timeout 10m bash -c 'until oc --kubeconfig /root/ibi-cluster/kubeconfig get clusterversion,clusteroperators; do sleep 5; done'
```

**Review by eranco74 — COMMENTED** [review]
/approve
just nit

File: ci-operator/step-registry/image-based-install-operator/post/delete-ici/image-based-install-operator-post-delete-ici-commands.sh (hunk 2 above)
Comment (giladravid16) [reply ci-operator/step-registry/image-based-install-operator/post/delete-ici/image-based-install-operator-post-delete-ici-commands.sh:28]: From the logs I saw it takes exactly 10 minutes, so maybe 20 minutes timeout?

**Pull request merged**